src/libraries/System.Private.CoreLib/gen/IntrinsicsInSystemPrivateCoreLibAnalyzer.cs

```diff
@@ -0,0 +1,693 @@
+                        {
+                            if (member.Kind == SymbolKind.Property)
+                            {
+                                if (member.Name == "IsSupported")
+                                {
+                                    namedTypesToBeProtected.Add(typeSymbol);
+                                }
+                            }
+                        }
+                    }
+                }
+
+                context.RegisterSymbolStartAction(context =>
+                {
+                    var methodSymbol = (IMethodSymbol)context.Symbol;
+
+                    foreach (var attributeData in methodSymbol.GetAttributes())
+                    {
+                        if (bypassReadyToRunAttribute != null)
+                        {
+                            if (attributeData.AttributeClass.Equals(bypassReadyToRunAttribute, SymbolEqualityComparer.Default))
+                            {
+                                // This method isn't involved in ReadyToRun, and so doesn't need analysis
+                                return;
+                            }
+                        }
+                    }
+
+                    context.RegisterOperationAction(context =>
+                    {
+                        AnalyzeOperation(context.Operation, methodSymbol, context, onLoadData);
+                    },
+                    OperationKind.Invocation, OperationKind.PropertyReference);
+                }, SymbolKind.Method);
+            });
+        }
+
+        private static ISymbol? GetOperationSymbol(IOperation operation)
+            => operation switch
+            {
+                IInvocationOperation iOperation => iOperation.TargetMethod,
+                IMemberReferenceOperation mOperation => mOperation.Member,
+                _ => null,
+            };
+
+        private static INamedTypeSymbol? GetIsSupportedTypeSymbol(SemanticModel model, MemberAccessExpressionSyntax memberAccessExpression)
+        {
+            if (memberAccessExpression.Name is IdentifierNameSyntax identifierName && identifierName.Identifier.Text == "IsSupported")
+            {
+                var symbolInfo = model.GetSymbolInfo(memberAccessExpression);
+                return symbolInfo.Symbol.ContainingSymbol as INamedTypeSymbol;
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        private static INamedTypeSymbol? GetIsSupportedTypeSymbol(SemanticModel model, IdentifierNameSyntax identifierName)
+        {
+            var symbolInfo = model.GetSymbolInfo(identifierName);
+
+            if (identifierName.Identifier.Text == "IsSupported")
+                return symbolInfo.Symbol.ContainingSymbol as INamedTypeSymbol;
+            else
+                return null;
+        }
+
+        private static INamedTypeSymbol[] GatherAndConditions(SemanticModel model, ExpressionSyntax expressionToDecompose)
```

Comment:
This would be better done operating on `IOperation` instances than syntax, but that can be a later refactoring (as this does work).